pkg/apis/duck/v1beta1/subscribable_types.go

```diff
@@ -0,0 +1,177 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"knative.dev/pkg/apis"
+	"knative.dev/pkg/apis/duck"
+)
+
+// +genduck
+
+// Subscribable is the schema for the subscribable portion of the spec
+// section of the resource.
+type Subscribable struct {
+	// This is the list of subscriptions for this subscribable.
+	// +patchMergeKey=uid
+	// +patchStrategy=merge
+	Subscribers []SubscriberSpec `json:"subscribers,omitempty" patchStrategy:"merge" patchMergeKey:"uid"`
+}
+
+// Subscribable is an Implementable "duck type".
+var _ duck.Implementable = (*Subscribable)(nil)
+
+// SubscriberSpec defines a single subscriber to a Subscribable.
+// Ref is a reference to the Subscription this SubscriberSpec was created for
+// SubscriberURI is the endpoint for the subscriber
+// ReplyURI is the endpoint for the reply
+// At least one of SubscriberURI and ReplyURI must be present
+type SubscriberSpec struct {
+	// UID is used to understand the origin of the subscriber.
+	// +optional
+	UID types.UID `json:"uid,omitempty"`
+	// Generation of the origin of the subscriber with uid:UID.
+	// +optional
+	Generation int64 `json:"generation,omitempty"`
+	// +optional
+	SubscriberURI *apis.URL `json:"subscriberUri,omitempty"`
+	// +optional
+	ReplyURI *apis.URL `json:"replyUri,omitempty"`
```

Comment:
`replyURI` ?

Reply:
Nope, these got changed to these. These are the cases we should be using for the JSON names. @n3wscott